File: framework/src/main/java/org/example/loader/PageDispatcher.java
package org.example.loader;

import org.example.core.intent.IIntent;
import org.example.core.dispatcher.IPageDispatcher;
import org.example.core.loader.IPageLoader;

class PageDispatcher implements IPageDispatcher {
    private IPageLoader pageLoader;

    @Override
    public void register(IPageLoader pageLoader) {
        this.pageLoader = pageLoader;
    }

    @Override
    public void dispatch(IIntent intent) {
        pageLoader.show(intent);
    }
}

File: framework/src/main/kotlin/org/example/loader/PageDispatcher.kt
package org.example.loader

import org.example.core.dispatcher.IPageDispatcher
import org.example.core.intent.IIntent
import org.example.core.loader.IPageLoader

internal class PageDispatcher : IPageDispatcher {
    private var pageLoader: IPageLoader? = null

    override fun register(pageLoader: IPageLoader?) {
        this.pageLoader = pageLoader
    }

    override fun dispatch(intent: IIntent?) {
        pageLoader?.show(intent)
    }
}
